Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.h

```diff
@@ -97,6 +97,10 @@ class CLuaEngineDefs : public CLuaDefs
 
     static void EnginePreloadWorldArea(CVector position, std::optional<PreloadAreaOption> option);
 
+    static bool EngineFramerateFixingResetProperties();
+    static bool EngineFramerateFixingSetProperty(std::string strPropertyName, float timestep);
```

Comment:
These functions should be void (returning nil), since they always return true anyway. There’s no reason for them to be bool.

Reply:
these are lua functions, should return true if function executes successfully

Reply:
> these are lua functions, should return true if function executes successfully

In this case, the function always completes successfully, so it doesn’t need to return true, because using it in a condition makes no sense. With the new parser, all new functions that always return true are simply void and return nil. There’s no reason for a function to be bool if it always returns the same value.

Reply:
> > these are lua functions, should return true if function executes successfully
> 
> In this case, the function always completes successfully, so it doesn’t need to return true, because using it in a condition makes no sense. With the new parser, all new functions that always return true are simply void and return nil. There’s no reason for a function to be bool if it always returns the same value.

ooh, that makes sense. I didn't know that. Thanks

Reply:
> > these are lua functions, should return true if function executes successfully
> 
> In this case, the function always completes successfully, so it doesn’t need to return true, because using it in a condition makes no sense. With the new parser, all new functions that always return true are simply void and return nil. There’s no reason for a function to be bool if it always returns the same value.

But I found it returns nil instead of "true" when it success.

Reply:
And that’s exactly how it should be.

Reply:
> And that’s exactly how it should be.

oh, ok. now I fixed everything